Dashboard/Connectors/Csv/CsvConnector.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Dashboard.Models;
+using Dashboard.Serialisation.Csv;
+using Dashboard.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Dashboard.Connectors.Csv;
+
+public class CsvConnector(ILogger<CsvConnector> logger) : IConnector
+{
+    public event EventHandler<GpsData>? GpsDataUpdated;
+    public event EventHandler<AvData>? AvDataUpdated;
+    public event EventHandler<ResData>? ResDataUpdated;
+    public event EventHandler<RawData>? RawDataUpdated;
+    public event EventHandler<bool>? HeartBeatUpdated;
+
+    private Thread? _readThread;
+    private StreamReader? _reader;
+    private bool _shouldStop;
+
+    public void Start(IConnectorArgs args)
+    {
+        if (args is not CsvConnectorArgs csvArgs)
+        {
+            logger.LogError("Invalid arguments passed to Csv Connector");
+            return;
+        }
+
+        // Open the file
+        _reader = OpenReader(csvArgs.FilePath);
+        if (_reader == null)
+        {
+            logger.LogError("Failed to open log replay file");
+            return;
+        }
+
+        // Validate the file header
+        var header = _reader.ReadLine();
+        if (header != CsvDataSerialiser.Header)
+        {
+            logger.LogError("Invalid file header");
+            return;
+        }
+
+        _readThread = new Thread(() =>
+        {
+            logger.LogInformation("Starting CSV read thread");
+
+            while (!_shouldStop)
+            {
+                var line = _reader.ReadLine();
+
+                if (line == null)
+                {
+                    // We are at EOF, loop back to the start of the file
+                    _reader.BaseStream.Seek(0, SeekOrigin.Begin);
+                    // and skip the header
+                    _reader.ReadLine();
+
+                    continue;
+                }
+
+                ParseData(line);
+
+                // Sleep for a bit
+                Thread.Sleep(1000);
```

Comment:
how come sleep

Reply:
so we don't spam the app, updates the data every second